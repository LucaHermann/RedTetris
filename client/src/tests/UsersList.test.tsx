import React from "react";
import { configure, shallow } from "enzyme";
import * as ReactSixteenAdapter from "enzyme-adapter-react-16";
import CharacterList from "../components/RoomSettings/UsersList";

// tslint:disable: quotemark

const adapter = ReactSixteenAdapter as any;
configure({ adapter: new adapter.default() });

describe("<CharacterList />", () => {
  let props: any;
  let wrapper: any;
  beforeEach(() => {
    props = {
      me: {
        id: "1",
        userName: "1",
        roomId: "1",
      },
      room: {
        id: "1",
        inGame: false,
        players: [
          {
            id: "1",
            userName: "1",
            roomId: "1",
          },
          {
            id: "2",
            userName: "2",
            roomId: "1",
          },
        ],
        gameMaster: {
          id: "1",
          userName: "1",
          roomId: "1",
        },
        isPrivate: false,
      },
      leaveRoomHandler: jest.fn(),
    };
    wrapper = shallow(<CharacterList {...props} />);
  });

  it("Should render a <List> with playrs in the room", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Should leave the room when icon clicked", () => {
    wrapper
      .find('Icon[name="close"]')
      // tslint:disable-next-line: no-empty
      .simulate("click", { preventDefault() {} });
    expect(props.leaveRoomHandler).toBeCalled();
  });
});
