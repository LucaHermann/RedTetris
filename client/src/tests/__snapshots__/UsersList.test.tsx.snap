// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterList /> Should render a <List> with playrs in the room 1`] = `"<div class=\\"userListStyle\\"><div class=\\"ui segment\\"><div role=\\"list\\" class=\\"ui list Charlist\\"><h3 style=\\"color:#31E981\\">Players</h3><div role=\\"listitem\\" class=\\"item\\"><div class=\\"content\\"><i aria-hidden=\\"true\\" class=\\"yellow star small circular icon\\"></i></div><div style=\\"color:#000000\\" class=\\"content\\">1</div><div class=\\"content\\"><i aria-hidden=\\"true\\" class=\\"red close link icon\\"></i></div></div><div role=\\"listitem\\" class=\\"item\\"><div class=\\"content\\"><i aria-hidden=\\"true\\" class=\\"teal user small circular icon\\"></i></div><div style=\\"color:#000000\\" class=\\"content\\">2</div><div class=\\"content\\"></div></div></div></div></div>"`;
